lib/StaticUniformGrid.js

```diff
@@ -0,0 +1,277 @@
+        grid : this,
+        item : undefined
+    };
+
+    //For each object:
+    for (i = 0; i < objectCount; i++) {
+        var item = data[i];
+
+        // Get the object's AABB
+        minScratch.x = Number.POSITIVE_INFINITY;
+        minScratch.y = Number.POSITIVE_INFINITY;
+        minScratch.z = Number.POSITIVE_INFINITY;
+        maxScratch.x = Number.NEGATIVE_INFINITY;
+        maxScratch.y = Number.NEGATIVE_INFINITY;
+        maxScratch.z = Number.NEGATIVE_INFINITY;
+        objectAABBFunction(item, minScratch, maxScratch);
+
+        // Step over the cells in the AABB, checking for each cell if the object intersects this cell
+        parameters.item = item;
+        StaticUniformGrid.boundingBoxMarch(this, minScratch, maxScratch, binObject, parameters);
+    }
+
+    // Store all the object copies in one contiguous array for better spatial locality
+    var cellItems = this.data;
+    var cellIndices = this.cellIndices;
+    var cellCounts = this.cellCounts;
+
+    var firstFreeIndex = 0;
+    for (i = 0; i < cellCount; i++) {
+        var cellData = allCellData[i];
+        var itemCount = cellData.itemCount;
+        var items = cellData.items;
+        if (itemCount > 0) {
+            cellIndices.push(firstFreeIndex);
+        } else {
+            cellIndices.push(-1);
+        }
+        cellCounts.push(itemCount);
+        firstFreeIndex += itemCount;
+        for (var j = 0; j < itemCount; j++) {
+            cellItems.push(clone(items[j]));
+        }
+    }
+}
+
+/**
+ * Function for executing a function on each cell touched by an axis aligned bounding box within the uniform grid.
+ *
+ * @param {StaticUniformGrid} [grid] A StaticUniformGrid to be marched over
+ * @param {Cartesian3} [min] The minimum coordinate of the AABB.
+ * @param {Cartesian3} [max] The maximum coordinate of the AABB.
+ * @param {Function} [marchFunction] A function to apply at each cell. The function should expect as inputs its parameters, the cell's minimum coordinate, and the cell width.
+ * @param {Object} [parameters] Parameters for the marchFunction
+ */
+
+var cellMarchScratch = new Cartesian3();
+StaticUniformGrid.boundingBoxMarch = function(grid, min, max, marchFunction, parameters) {
+    var stepWidth = grid.cellWidth;
+    var gridMin = grid.AABB.minimum;
+
+    // Compute the minimum coordinate of the first cell
+    var xStart = Math.floor((min.x - gridMin.x) / stepWidth) * stepWidth + gridMin.x;
+    var yStart = Math.floor((min.y - gridMin.y) / stepWidth) * stepWidth + gridMin.y;
+    cellMarchScratch.z = Math.floor((min.z - gridMin.z) / stepWidth) * stepWidth + gridMin.z;
+
+
+    // Compute the number of cells that min and max cover in each dimension.
+    var xCount = Math.floor((max.x - min.x) / stepWidth) + 1;
+    var yCount = Math.floor((max.y - min.y) / stepWidth) + 1;
+    var zCount = Math.floor((max.z - min.z) / stepWidth) + 1;
+
+    // March over the cells that the grid covers.
+    for (var z = 0; z < zCount; z++) {
```

Comment:
In practice, I don't know if it matters anymore, but I always write loops with, for example, `++z` so it saves a potential copy.